rest/oidc_api.go

```diff
@@ -237,6 +237,13 @@ func (h *handler) getOIDCCallbackURL() string {
 		base.Warnf("Can't calculate OIDC callback URL without DB in path.")
 		return ""
 	} else {
-		return fmt.Sprintf("%s://%s/%s/%s", scheme, h.rq.Host, dbName, "_oidc_callback")
+		callbackURL := fmt.Sprintf("%s://%s/%s/_oidc_callback", scheme, h.rq.Host, dbName)
+		if !isDefault && providerName != "" {
+			if callbackURL, err := auth.SetURLQueryParam(callbackURL, auth.OIDCAuthProvider, providerName); err != nil {
```

Comment:
This `callbackURL` variable is being rescoped inside the ifs, and won't be available when we return on line 246.

E.g: https://play.golang.org/p/3lgxF2-vZzq

I think we still need to add an end-to-end test to verify we're actually performing the correct behaviour on this callback URL, or at least some level of manual testing/verification.

Reply:
Great catch! the provider name was not getting added to the newly generated callback URL when no callback URL is specified in the configuration due to scope change of callbackURL inside the 'if' block. The issue has been fixed and added a Unit test which validate the functionality.